### PR TITLE
Button: VR Certification Fixes

### PR DESCRIPTION
**Pull request by eniomoura**
#### What changed?

Fixed spacing for ButtonGroup and spacing between icons and text in Button

#### Why?

Requested by the Visual Refresh Design Certification Team

### Links

- [Jira](https://jira.pinadmin.com/browse/CPW-7399)

### CODE REVIEW

File: packages/gestalt/src/Button.tsx [hunk 1]
```diff
@@ -120,6 +120,23 @@ type Props = {
...
 };
 
+function InternalButtonWrapper({ children }: { children: ReactNode }) {
+  const isInVRExperiment = useInExperiment({
+    webExperimentName: 'web_gestalt_visualRefresh',
+    mwebExperimentName: 'web_gestalt_visualRefresh',
+  });
+
+  return isInVRExperiment ? (
+    <div style={{ display: 'flex', alignItems: 'center', justifyContent: 'center', gap: 6 }}>
```
Comment (AlbertCarreras) [inline packages/gestalt/src/Button.tsx:130]: any reason to not use Flex?

Comment (AlbertCarreras) [reply packages/gestalt/src/Button.tsx:130]: oh, 6 px?

Comment (AlbertCarreras) [reply packages/gestalt/src/Button.tsx:130]: ill bring this to the meeting on tuesday

File: packages/gestalt/src/ButtonGroup.tsx [hunk 2]
```diff
@@ -16,15 +17,26 @@ type Props = {
...
 
   return (
     <Box display="flex" margin={-1} wrap>
-      {Children.map(children, (child) =>
-        child !== null && child !== undefined ? <Box padding={1}>{child}</Box> : null,
-      )}
+      {isInVRExperiment
+        ? Children.map(children, (child) =>
+            child !== null && child !== undefined ? (
+              <div style={{ padding: 2 }}>{child}</div>
```
Comment (AlbertCarreras) [inline packages/gestalt/src/ButtonGroup.tsx:34]: this is problematic because they don't even have tokens for 2 px!

Comment (eniomoura) [reply packages/gestalt/src/ButtonGroup.tsx]: Using Flex instead

**Comment by eniomoura** [comment]
Looking into focus issue
image

File: packages/gestalt/src/Button.tsx (hunk 1 above)
Comment (eniomoura) [reply packages/gestalt/src/Button.tsx:130]: Fixed with the new Box solution!

**Comment by eniomoura** [comment]
Also changed Icons example

File: packages/gestalt/src/Button.tsx [hunk 3]
```diff
@@ -120,6 +120,23 @@ type Props = {
   type?: 'button' | 'submit';
 };
 
+function InternalButtonWrapper({ children }: { children: ReactNode }) {
```
Comment (AlbertCarreras) [inline packages/gestalt/src/Button.tsx:123]: why this wrapper seems unnecessary?

File: packages/gestalt/src/ButtonGroup.tsx [hunk 4]
```diff
@@ -1,5 +1,7 @@
 import { Children, ReactNode } from 'react';
+import { Flex } from '.';
```
Comment (AlbertCarreras) [inline packages/gestalt/src/ButtonGroup.tsx:2]: Can we do this change in a separate PR? Just ButtonGroup?

File: packages/gestalt/src/Button.css [hunk 5]
```diff
@@ -53,6 +53,14 @@
   padding: var(--sema-space-150) var(--sema-space-300);
 }
 
+a.smVr:focus {
+  padding: var(--sema-space-100) calc(var(--sema-space-200) + 2px);
+}
+
+a.lgVr:focus {
+  padding: var(--sema-space-150) calc(var(--sema-space-300) + 2px);
```
Comment (AlbertCarreras) [inline packages/gestalt/src/Button.css:61]: what are these random  2px?

Comment (eniomoura) [reply packages/gestalt/src/Button.css:61]: Those are offsets for the focus inner border, to fix the shifting issue, I should have used tokens here instead though, will fix

File: packages/gestalt/src/ButtonGroup.tsx (hunk 4 above)
Comment (eniomoura) [reply packages/gestalt/src/ButtonGroup.tsx]: https://github.com/pinterest/gestalt/pull/3840

File: packages/gestalt/src/Button.css (hunk 5 above)
Comment (eniomoura) [reply packages/gestalt/src/Button.css]: DONE

**Review by AlbertCarreras — APPROVED** [review]
